Fix tests

Tests terminate with 
```
[ERROR] 
[ERROR] Please refer to /Users/micbinz/git/github/app_scream/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /Users/micbinz/git/github/app_scream && /opt/homebrew/Cellar/openjdk/19.0.2/libexec/openjdk.jdk/Contents/Home/bin/java -jar /Users/micbinz/git/github/app_scream/target/surefire/surefirebooter10054496065027517448.jar /Users/micbinz/git/github/app_scream/target/surefire 2023-03-06T20-40-47_294-jvmRun1 surefire9013713109967847294tmp surefire_018150433603433020099tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] Crashed tests:
[ERROR] de.michab.scream.RuntimeXTest
...
```

--- a/src/main/java/de/michab/scream/ScreamEvaluator.java
+++ b/src/main/java/de/michab/scream/ScreamEvaluator.java
@@ -89,7 +89,8 @@ public final class ScreamEvaluator implements ScriptEngine
     {
         return new PortIn(
                 "stdin",
-                _context.peek().getReader() );
+                _context.peek().getReader(),
+                false );
     }
 
     /**
@@ -101,7 +102,8 @@ public final class ScreamEvaluator implements ScriptEngine
     {
         return new PortOut(
                 "stdout",
-                _context.peek().getWriter() );
+                _context.peek().getWriter(),
+                false );
     }
 
     /**
@@ -113,7 +115,8 @@ public final class ScreamEvaluator implements ScriptEngine
     {
         return new PortOut(
                 "stderr",
-                _context.peek().getErrorWriter() );
+                _context.peek().getErrorWriter(),
+                false );
     }
 
     public Environment getInteraction()

--- a/src/main/java/de/michab/scream/fcos/Port.java
+++ b/src/main/java/de/michab/scream/fcos/Port.java
@@ -34,14 +34,47 @@ public abstract class Port
     private final Type _type;
 
     /**
+     * If {@code true} then the port is finalized.
+     */
+    private final boolean _finalize;
+
+    /**
      * Create a textual port.
      *
      * @param name The port's name.
      */
     public Port( String name )
     {
-        this( name, Type.Textual );
+        this( name, Type.Textual, true );
     }
+
+    /**
+     * Create a textual port.
+     *
+     * @param name The port's name.
+     * @param finalize If true, then the port is finalized.  If false then
+     * finalization is skipped.  For system in/out/err this should be false.
+     */
+    public Port( String name, boolean finalize )
+    {
+        this( name, Type.Textual, finalize );
+    }
+
+    /**
+     * Create a typed port.
+     *
+     * @param name The port's name.
+     * @param type The port's type.
+     * @param finalize If true, then the port is finalized.  If false then
+     * finalization is skipped.  For system in/out/err this should be false.
+     */
+    public Port( String name, Type type, boolean finalize )
+    {
+        _name = name;
+        _type = type;
+        _finalize = finalize;
+    }
+
     /**
      * Create a typed port.
      *
@@ -50,8 +83,7 @@ public abstract class Port
      */
     public Port( String name, Type type )
     {
-        _name = name;
-        _type = type;
+        this( name, type, true );
     }
 
     String name()
@@ -98,6 +130,8 @@ public abstract class Port
             throws
             Throwable
     {
+        if ( !_finalize )
+            return;
         close();
         // Chain finalisers.
         super.finalize();

--- a/src/main/java/de/michab/scream/fcos/PortIn.java
+++ b/src/main/java/de/michab/scream/fcos/PortIn.java
@@ -64,6 +64,11 @@ public class PortIn
         super( name );
         _inPort = in;
     }
+    public PortIn( String name, Reader in, boolean finalize )
+    {
+        super( name, finalize );
+        _inPort = in;
+    }
 
     /**
      * Create a named port.  The passed name is used for opening the file.

--- a/src/main/java/de/michab/scream/fcos/PortOut.java
+++ b/src/main/java/de/michab/scream/fcos/PortOut.java
@@ -43,6 +43,11 @@ public class PortOut
         super( name );
         _outPort = out;
     }
+    public PortOut( String name, Writer out, boolean finalize )
+    {
+        super( name, finalize );
+        _outPort = out;
+    }
 
     /**
      * Create a named port.  The passed name is used for opening the file.

